File: 351004/Bunevich/labs/task330/src/main/java/com/example/demo/models/Post.java
package com.example.demo.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "tbl_post", schema = "distcomp")
public class Post extends BaseEntity{
    @ManyToOne(optional = false)
    @JoinColumn(name = "story_id", nullable = false)
    private Story story;

    @Column(nullable = false, length = 2048)
    private String content;

    // Геттеры
    public Story getStory() {
        return story;
    }

    public String getContent() {
        return content;
    }

    // Сеттеры
    public void setStory(Story story) {
        this.story = story;
    }

    public void setContent(String content) {
        this.content = content;
    }
//    Writer → Story	один ко многим
//    Story → Post	один ко многим
//    Story ↔ Tag	многие ко многим
}

//Необходимо реализовать хранение сущностей в реляционной базе данных Postgres
//инициализации базы данных, рекомендуется использовать liquibase в формате XML

//обобщенный интерфейс для хранения и поиска данных сущностей для CRUD операций:
//операции поиска и выборки должны поддерживать пагинацию, фильтрацию и сортировку

File: 351003/Kisel/JPA/RV_Kisel_lab2_Task320/Models/Entities/Post.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RV_Kisel_lab2_Task320.Models.Entities;

[Table("tbl_post")]
public class Post {
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("content")]
    public string Content { get; set; } = string.Empty;

    [Column("news_id")]
    public int NewsId { get; set; }

    public News? News { get; set; }
}